src/transformers/models/mask2former/modeling_mask2former.py

```diff
@@ -0,0 +1,3393 @@
+        num_cur_levels = 0
+        
+        features = features[1:] 
+
+        for idx,feature in enumerate(features[::-1]):
+            print("features shape:", feature.shape)
+            input_embeds.append(self.input_projection[idx](feature))
+            position_embeddings.append(self.position_embedding_layer(feature))
+        
+        encoder_output: Mask2FormerMSDeformableAttnEncoderOutput = self.msdeformattn_transformer(input_embeds, position_embeddings)
+        
+        final_encoder_output = encoder_output.last_hidden_state
+        spatial_shapes = encoder_output.spatial_shapes
+        level_start_index = encoder_output.level_start_index
+        print('final_encoder_output:',final_encoder_output)
+        print("spatial_shapes:",spatial_shapes)
+        print("level_start_index:",level_start_index)
+        batch_size = final_encoder_output.shape[0]
+
+        split_size_or_sections = [None] * self.transformer_num_feature_levels
+
+        for idx in range(self.transformer_num_feature_levels):
+            if idx < self.transformer_num_feature_levels -1:
+                split_size_or_sections[idx] = level_start_index[idx + 1] - level_start_index[idx]
+
+            else:
+                split_size_or_sections[idx] = final_encoder_output.shape[1] - level_start_index[idx]
+
+        final_encoder_output = torch.split(final_encoder_output, split_size_or_sections, dim=1)
+        print("final_encoder_output:",final_encoder_output)
+
+        for idx, feature in enumerate(final_encoder_output):
+            feature = feature.transpose(1,2).view(batch_size, -1, spatial_shapes[idx][0], spatial_shapes[idx][1])
+            transformer_encoder_features.append(feature)
+
+        ##append `transformer_encoder_features` with extra FPN levels
+
+        for idx,feature in enumerate(features[:self.num_fpn_levels][::-1]):
+            lateral_conv = self.lateral_convs[idx]
+            output_conv = self.output_convs[idx]
+            cur_fpn = lateral_conv(feature)
+            upsampled_feature = nn.functional.interpolate(transformer_encoder_features[-1], size=cur_fpn.shape[-2:], mode='bilinear', align_corners=False)
+            output = cur_fpn + upsampled_feature
+            output = output_conv(output)
+            transformer_encoder_features.append(output)
+
+        for feature in transformer_encoder_features:
+            if num_cur_levels < self.mask2former_num_feature_levels:
+                multi_scale_features.append(feature)
+                num_cur_levels += 1
+
+        # we use the last feature map
+        last_feature_projected = self.mask_projection(transformer_encoder_features[-1])
+
+
+        return Mask2FormerPixelDecoderOutput(
+            last_hidden_state=last_feature_projected,
+            transformer_encoder_features=transformer_encoder_features[0],
+            multi_scale_features=multi_scale_features,
+            hidden_states=tuple(transformer_encoder_features)
+        )
```

Comment:
Please see my comments above, `multi_scale_features` is the `hidden_states`:
```suggestion
        return Mask2FormerPixelDecoderOutput(
            last_hidden_state=last_feature_projected,
            hidden_states=tuple(multi_scale_features)
        )
```

Reply:
sure fixing all changes related to this